### ISSUE TITLE
submitting MIR cert for unknown stake address leads to `Rewards differ between ledger and db-sync`

### ISSUE THREAD

**Issue by mkoura** [input-output-hk/cardano-db-sync#934]
**OS**
Your OS: Fedora 34

**Versions**
The `db-sync` version: `12.0.0-pre3`
PostgreSQL version: `11.11`

**Build/Install Method**
The method you use to build or install `cardano-db-sync`: `nix-build`

**Run method**
The method you used to run `cardano-db-sync` (eg Nix/Docker/systemd/none): none

**Additional context**
Latest master node running on local cluster, Alonzo era, Protocol Version 6.

**Problem Report**
I have a test that tries to send funds from reserves or treasury pot to a stake address that was never registered on the blockchain.
The issue cannot be reproduced when the address was previously registered.

First variant of the test tries to transfer 1500000 ADA from the treasury pot. At the epoch boundary, following is reported by db-sync:

```
[db-sync-node:Warning:58] [2021-10-29 10:47:05.62 UTC] validateEpochRewards: rewards spendable in epoch 24 expected total of 1094.074712 ADA but got 1501094.074712 ADA
[db-sync-node:Error:58] [2021-10-29 10:47:05.62 UTC] Rewards differ between ledger and db-sync.
Please report at https://github.com/input-output-hk/cardano-db-sync/issues.
```

Second variant of the test tries to transfer 50000000 ADA from reserves pot. At the epoch boundary, following is reported by db-sync:

```
[db-sync-node:Warning:58] [2021-10-29 10:47:06.20 UTC] validateEpochRewards: rewards spendable in epoch 29 expected total of 1117.391357 ADA but got 50001117.391357 ADA
[db-sync-node:Error:58] [2021-10-29 10:47:06.20 UTC] Rewards differ between ledger and db-sync.
Please report at https://github.com/input-output-hk/cardano-db-sync/issues.
```

From outputs above it seems db-sync thinks the amounts were actually transferred. In reality the MIR cert didn't have any effect, the stake addresses never existed on the blockchain.

**Comment by erikd** [comment]
This almost certainly has the same root cause as https://github.com/input-output-hk/cardano-db-sync/issues/921

**Comment by erikd** [comment]
Fixed in PR #944 .
